smartsim/settings/arguments/launch/slurm.py

```diff
@@ -44,27 +44,45 @@
 
 
 def _as_srun_command(
-    args: ShellLaunchArguments,
+    args: SlurmLaunchArguments,
     exe: t.Sequence[str],
     path: pathlib.Path,
     env: EnvironMappingType,
     stdout_path: pathlib.Path,
     stderr_path: pathlib.Path,
-) -> ShellLauncherCommand:
-    command_tuple = (
-        "srun",
+) -> SrunCommand:
+    # TODO: Should probably use the `SmartSimEntity` name here, but right now
+    #       there is no way to access it. In the meantime, we will just use the
+    #       exe being called.
+    app, *_ = exe
+    try:
+        _, job_name = app.rsplit(os.path.sep, maxsplit=1)
+    except ValueError:
+        job_name = app
```

Comment:
`ValueError` seems broad, consider
```suggestion
    *_, job_name = app.rsplit(os.path.sep, maxsplit=1)
```